src/components/implementation/tests/unit_cxx/cxx.h

```diff
@@ -0,0 +1,175 @@
+#include <vector>
+#include <algorithm>
+#include <set>
+#include <queue>
+#include <map>
+#include <string>
+#include <iostream>
+using namespace std;
+#ifdef __cplusplus
+extern "C" {
+#include <print.h>
+#endif
+
+#ifdef __cplusplus
+}
+#endif
+
+class Global_foo {
+public:
+	Global_foo();
+	~Global_foo();
+};
+Global_foo::Global_foo()
+{
+//	cout<<"Global foo constructor"<<endl;
+	printc("Global foo constructor\n");
+}
+Global_foo::~Global_foo()
+{
+	cout<<"Global foo destructor"<<endl;
+}
+
+class Global_bar {
+public:
+	Global_bar();
+//	~Global_bar();
+};
+Global_bar::Global_bar()
+{
+//	cout<<"Global bar constructor"<<endl;
+	printc("Global bar constructor\n");
+}
+/* Global_bar::~Global_bar() */
+/* { */
+/* 	printc("Global bar destructor\n"); */
+/* } */
+
+class cl {
+	int i;
+public:
+	cl(void);
+	cl(int a);
+	~cl();
+	int get_i();
+	void put_i(int j);
+};
+cl::cl(void)
+{
+	i = 0;
+}
+cl::cl(int a)
+{
+	i = a;
+	cout<<"create object "<<i<<endl;
+}
+cl::~cl()
+{
+	cout<<"destroy object "<<i<<endl;
+}
+int cl::get_i()
```

Comment:
function return type should be on the previous line.  We likely want blank lines between methods.